File: src/main/java/name/nkid00/rcutil/helper/TextHelper.java
package name.nkid00.rcutil.helper;

import net.minecraft.text.LiteralText;
import net.minecraft.text.MutableText;
import net.minecraft.text.Text;
import net.minecraft.text.TranslatableText;
import net.minecraft.util.Formatting;

public class TextHelper {
    public static MutableText copy(Text text) {
        return text.copy();
    }

    public static MutableText empty() {
        return literal("");
    }

    public static MutableText literal(String string) {
        return new LiteralText(string);
    }

    public static MutableText translatable(String key, Object... args) {
        return new TranslatableText(key, args);
    }

    public static MutableText formatted(Text text, Formatting formatting) {
        return empty().append(text).formatted(formatting);
    }

    public static MutableText info(Text text) {
        return text.copy();
    }

    public static MutableText warn(Text text) {
        return formatted(text, Formatting.YELLOW);
    }

    public static MutableText error(Text text) {
        return formatted(text, Formatting.RED);
    }
}

File: src/main/java/me/nk0/rcu/helper/TextHelper.kt
package me.nk0.rcu.helper

import net.minecraft.text.MutableText
import net.minecraft.text.Text
import net.minecraft.util.Formatting

object TextHelper {
    fun copy(text: Text): MutableText {
        return text.copy()
    }

    fun empty(): MutableText {
        return Text.empty()
    }

    @JvmStatic
    fun literal(string: String?): MutableText {
        return Text.literal(string)
    }

    @JvmStatic
    fun translatable(key: String?, vararg args: Any?): MutableText {
        return Text.translatable(key, *args)
    }

    @JvmStatic
    fun formatted(text: Text?, formatting: Formatting?): MutableText {
        return empty().append(text).formatted(formatting)
    }

    fun info(text: Text): MutableText {
        return text.copy()
    }

    @JvmStatic
    fun warn(text: Text?): MutableText {
        return formatted(text, Formatting.YELLOW)
    }

    @JvmStatic
    fun error(text: Text?): MutableText {
        return formatted(text, Formatting.RED)
    }
}
